### PR TITLE
Feature/ape rewards

### PR DESCRIPTION
**Pull request by arnabmitra**
## Description
addresses https://github.com/provenance-io/provenance/issues/626
Closes: https://github.com/provenance-io/provenance/issues/626

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by nullpointer0x00 — APPROVED** [review]
LGTM

**Review by Taztingo — APPROVED** [review]
LGTM

**Review by nullpointer0x00 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,8 +38,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 # Provenance Specific Releases
 
 ## Unreleased
-
-* nothing
+* Add support for event data injection into end block handlers (https://github.com/provenance-io/provenance/issues/626)
```
Comment (SpicyLemon) [inline CHANGELOG.md:41]: Nit: Needs a subsection heading here, e.g. `### Features`

Comment (arnabmitra) [reply CHANGELOG.md:44]: good find..changed

File: baseapp/baseapp_test.go [hunk 2]
```diff
@@ -192,6 +193,12 @@ func setupBaseAppWithSnapshots(t *testing.T, blocks uint, blockTxs int, options
 	return app, teardown
 }
 
+func shouldPanic(t *testing.T, msg string, f func()) {
+	defer func() { recover() }()
+	f()
+	t.Errorf(msg)
+}
+
```
Comment (SpicyLemon) [inline baseapp/baseapp_test.go:201]: For future reference, there's a `.Panics` in assert/require. Also `.PanicsWithError` and `.NotPanics`.

**Review by SpicyLemon — APPROVED** [review]
LGTM.

Ran tests locally and they all passed.

**Comment by SpicyLemon** [comment]
Make sure to squash and merge. It'll make it easier to apply this to v0.46.

File: baseapp/baseapp_test.go (hunk 2 above)
Comment (arnabmitra) [reply baseapp/baseapp_test.go:201]: good points

**Pull request merged**